sklearn/preprocessing/tests/test_data.py

```diff
@@ -851,6 +853,118 @@ def test_robust_scaler_iris_quantiles():
+                      [-2, 4, 0, 0, 6, 8, 0, 10, 0, 0],
+                      [0, 0, 2.6, 4.1, 0, 0, 2.3, 0, 9.5, 0.1]]).T
+    X_neg = sparse.csc_matrix(X_neg)
+
+    normalizer = QuantileNormalizer()
+    assert_raises_regex(ValueError, "QuantileNormalizer only accepts "
+                        "non-negative sparse matrices", normalizer.fit, X_neg)
+    normalizer.fit(X)
+    assert_raises_regex(ValueError, "QuantileNormalizer only accepts "
+                        "non-negative sparse matrices",
+                        normalizer.transform, X_neg)
+    assert_raises_regex(ValueError, "QuantileNormalizer only accepts "
+                        "non-negative sparse matrices",
+                        normalizer.inverse_transform, X_neg)
+
+    X_bad_feat = np.array([[0, 25, 50, 0, 0, 0, 75, 0, 0, 100],
+                           [0, 0, 2.6, 4.1, 0, 0, 2.3, 0, 9.5, 0.1]]).T
+    assert_raises_regex(ValueError, "X does not have the same number of "
+                        "feature than the previously fitted data.",
+                        normalizer.transform, X_bad_feat)
+    assert_raises_regex(ValueError, "X does not have the same number of "
+                        "feature than the previously fitted data.",
+                        normalizer.inverse_transform, X_bad_feat)
+
+
+def test_quantile_normalizer_dense_toy():
+    X = np.array([[0, 25, 50, 75, 100],
+                  [2, 4, 6, 8, 10],
+                  [2.6, 4.1, 2.3, 9.5, 0.1]]).T
+
+    normalizer = QuantileNormalizer()
+    normalizer.fit(X)
+
+    X_trans = normalizer.fit_transform(X)
+    assert_almost_equal(np.min(X_trans, axis=0), 0.)
+    assert_almost_equal(np.max(X_trans, axis=0), 1.)
+
+    X_test = np.array([
+        [-1,  1,  0],
+        [101, 11, 10],
+    ])
+    expected = np.array([
+        [0, 0, 0],
+        [1, 1, 1],
+    ])
+    assert_array_almost_equal(normalizer.transform(X_test), expected)
+
+    X_trans_inv = normalizer.inverse_transform(X_trans)
+    assert_array_almost_equal(X, X_trans_inv)
+
+    # test subsampling
+    # FIXME: there is not comparison for the moment
+    random_state = 42
+    normalizer.set_params(**{'subsample': 3,
+                             'n_quantiles': 2,
+                             'random_state': random_state})
+    X_trans = normalizer.fit_transform(X)
+
+
+def test_quantile_normalizer_sparse_toy():
+    X = np.array([[0, 25, 50, 0, 0, 0, 75, 0, 0, 100],
+                  [2, 4, 0, 0, 6, 8, 0, 10, 0, 0],
+                  [0, 0, 2.6, 4.1, 0, 0, 2.3, 0, 9.5, 0.1]]).T
+    X = sparse.csc_matrix(X)
+
+    normalizer = QuantileNormalizer()
+    normalizer.fit(X)
+
+    X_trans = normalizer.fit_transform(X)
+    assert_array_almost_equal(np.min(X_trans.toarray(), axis=0), 0.)
+    assert_array_almost_equal(np.max(X_trans.toarray(), axis=0), 1.)
+
+    X_trans_inv = normalizer.inverse_transform(X_trans)
+    assert_array_almost_equal(X.toarray(), X_trans_inv.toarray())
```

Comment:
Please also compare that you get similar results when using `normalizer_dense = QuantileNormalizer().fit(X.toarray())`.